campaignion_foundation.js

```diff
@@ -7,10 +7,24 @@
 // Theme specific JS.
 Drupal.behaviors.campaignion_foundation = {};
 Drupal.behaviors.campaignion_foundation.attach = function (context, settings) {
-  // Fix for file upload with AJAX enabled.
-  // See https://www.drupal.org/project/drupal/issues/1513200
-  if (Drupal.file) {
-    $('input.form-submit', context).unbind('mousedown', Drupal.file.disableFields);
+  // Upload managed files on forms immediately when selected.
+  if (settings.file && settings.file.elements) {
+    $.each(settings.file.elements, function(selector) {
+      $(selector, context).on('change', function() {
+        var $input = $(this);
+        // Wait for validation to finish.
+        setTimeout(function () {
+          var $error = $('.file-upload-js-error');
+          if ($error.length) {
+            $error.addClass('callout alert');
+          }
+          else if ($input.val()) {
+            $input.closest('form').submit();
+            $input.siblings('label.button').attr('disabled', 'disabled');
+          }
+        }, 100);
```

Comment:
Using a `setTimout()` here seems like it would break due to race conditions. Is there another event that is emitted after the validation that we could use?

Reply:
Not that I know of. `validateExtension()` is triggered by a change event on the input (I’ve copied the selectors from there) and either adds a hardcoded error message or not. No events or Promises. See https://git.drupalcode.org/project/drupal/-/blob/7.x/modules/file/file.js#L66

On the other hand the function is so simple that I’m fairly confident it finishes in a few milliseconds.